File: 0624-maximum-distance-in-arrays/0624-maximum-distance-in-arrays.cpp
#include <vector>
#include <cstdlib>
#include <iostream>
#include <algorithm>  

using namespace std;

class Solution {
public:
    int maxDistance(vector<vector<int>>& arrays) {
        int n = arrays.size();
        int min_val = arrays[0][0];  
        int max_val = arrays[0].back();  
        int max_diff = 0;
        
        for(int i = 1; i < n; ++i) {
            max_diff = max(max_diff, arrays[i].back() - min_val);
        
            max_diff = max(max_diff, max_val - arrays[i][0]);
            
            
            min_val = min(min_val, arrays[i][0]);
            max_val = max(max_val, arrays[i].back());
        }
        
        return max_diff;
    }
};

File: 0624-maximum-distance-in-arrays/0624-maximum-distance-in-arrays.py
class Solution:
    def maxDistance(self, arrays: List[List[int]]) -> int:
        n = len(arrays)
        min_val = arrays[0][0]  
        max_val = arrays[0][-1]  
        max_diff = 0  
        
        for i in range(1, n):
            max_diff = max(max_diff, abs(arrays[i][-1] - min_val))
            
            max_diff = max(max_diff, abs(max_val - arrays[i][0]))
            
            
            min_val = min(min_val, arrays[i][0])
            max_val = max(max_val, arrays[i][-1])
        
        return max_diff
         
        
       
    
                
                
        